Add: 32bpp SSE2 blitter palette animator

JGRennison opened the pull request:
This is a follow on from issue #6469.

Add an implementation of a 32bpp SSE2 blitter palette animator.
Create a new blitter mode: 32bpp-sse2-anim, which is 32bpp-anim + this.
32bpp-sse2-anim is now used by default where 32bpp-anim would have been.
Also use this with the 32bpp-sse4-anim blitter.

This doesn't include likely/unlikely hints (see discussion in #6469).
Blitter_32bppBase::ReallyAdjustBrightness hasn't currently been renamed, as it's the direct equivalent of ReallyAdjustBrightness in 32bpp_sse_func.hpp (see discussion in #6469).

PeterN commented on src/blitter/32bpp_base.cpp:
```diff
@@ -143,6 +143,36 @@ void Blitter_32bppBase::PaletteAnimate(const Palette &palette)
+						 r >= 255 ? 255 : min(r + ob * (255 - r) / 256, 255),
+						 g >= 255 ? 255 : min(g + ob * (255 - g) / 256, 255),
+						 b >= 255 ? 255 : min(b + ob * (255 - b) / 256, 255),
+						 colour.a);
```
Indentation is wild!

PeterN commented on src/blitter/32bpp_base.cpp:
```diff
@@ -143,6 +143,36 @@ void Blitter_32bppBase::PaletteAnimate(const Palette &palette)
+
+	uint16 r = GB(combined, 39, 9);
+	uint16 g = GB(combined, 23, 9);
+	uint16 b = GB(combined, 7, 9);
```
Sneaky way of combining 3 multiplications into 1, I like it.

JGRennison committed:
Add: 32bpp SSE2 blitter palette animator

Create a new blitter mode: 32bpp-sse2-anim, which is 32bpp-anim + this.
32bpp-sse2-anim is now used by default where 32bpp-anim would have been.
Also use this with the 32bpp-sse4-anim blitter.

See issue #6469.

JGRennison commented:
I've amended the commit with more sensible indentation as per your review, I hadn't noticed that.

PeterN commented on src/gfxinit.cpp:
```diff
@@ -284,6 +284,9 @@ static bool SwitchNewGRFBlitter()
+#ifdef WITH_SSE
+		{ "32bpp-sse2-anim", 1,  8, 32,  8, 32 },
+#endif
 		{ "32bpp-anim",      1,  8, 32,  8, 32 },
```
Does the order of this list matter? Can it go directly after 32bpp-sse4-anim without the extra WITH_SSE guard?

JGRennison replied:
The order does matter, the implementation selects the first matching and available blitter.

At minimum, 32bpp-sse2-anim should appear after 8bpp-optimized, and before 32bpp-anim.

PeterN replied:
Okay, that makes sense.

PeterN reviewed (approved):
Well documented, good work

andythenorth commented:
I tested it.  It's "snappier" :P
